libfwupd/fwupd-client.c

```diff
@@ -290,13 +290,17 @@ fwupd_client_parse_devices_from_variant (GVariant *val)
 	for (guint i = 0; i < sz; i++) {
 		FwupdDevice *dev;
 		g_autoptr(GVariant) data = NULL;
+		const gchar *dev_id;
 		data = g_variant_get_child_value (untuple, i);
 		dev = fwupd_device_from_variant (data);
 		if (dev == NULL)
 			continue;
 		g_ptr_array_add (array, dev);
+		dev_id = fwupd_device_get_id (dev);
+		if (dev_id == NULL)
```

Comment:
I'd much rather have the hash insertion protected by the `if` -- you also don't need to "cache" fwupd_device_get_id as it's just two pointer derefs. I'll fix things up, thanks.